Sources/AudioKit/Taps/FFTTap.swift

```diff
@@ -35,75 +44,126 @@ open class FFTTap: BaseTap {
-                // And then pack the input into the complex buffer (output)
-                transferBuffer.withUnsafeBufferPointer { pointer in
-                    pointer.baseAddress!.withMemoryRebound(to: DSPComplex.self,
-                                                           capacity: transferBuffer.count) {
-                        vDSP_ctoz($0, 2, &output, 1, vDSP_Length(inputCount))
-                    }
-                }
-
-                // Perform the FFT
-                vDSP_fft_zrip(fftSetup!, &output, 1, log2n, FFTDirection(FFT_FORWARD))
-
-                var magnitudes = [Float](repeating: 0.0, count: inputCount)
-                vDSP_zvmags(&output, 1, &magnitudes, 1, vDSP_Length(inputCount))
-
-                var scaledMagnitudes = [Float](repeating: 0.0, count: inputCount)
-
-                // Scale appropriate to the algorithm - results in strictly negative amplitude values (tested against Ableton Live's Spectrum Analyzer)
-                var scaleMultiplier = [Float(1.0 / Double(frameCount))]
-
-                if isNormalized {
-                    // Normalising
-                    scaleMultiplier = [1.0 / (magnitudes.max() ?? 1.0)]
-                }
-
-                vDSP_vsmul(&magnitudes,
-                           1,
-                           &scaleMultiplier,
-                           &scaledMagnitudes,
-                           1,
-                           vDSP_Length(inputCount))
-                
-                vDSP_destroy_fftsetup(fftSetup)
-                return scaledMagnitudes
+        var output = DSPSplitComplex(repeating: 0, count: inputCount)
+
+        let windowSize = bufferSizePOT
+        var transferBuffer = [Float](repeating: 0, count: windowSize)
+        var window = [Float](repeating: 0, count: windowSize)
+
+        // Hann windowing to reduce the frequency leakage
+        vDSP_hann_window(&window, vDSP_Length(windowSize), Int32(vDSP_HANN_NORM))
+        vDSP_vmul((buffer.floatChannelData?.pointee)!, 1, window,
+                  1, &transferBuffer, 1, vDSP_Length(windowSize))
+
+        // Transforming the [Float] buffer into a UnsafePointer<Float> object for the vDSP_ctoz method
+        // And then pack the input into the complex buffer (output)
+        transferBuffer.withUnsafeBufferPointer { pointer in
+            pointer.baseAddress!.withMemoryRebound(to: DSPComplex.self,
+                                                   capacity: transferBuffer.count) {
+                vDSP_ctoz($0, 2, &output, 1, vDSP_Length(inputCount))
             }
         }
+
+        // Perform the FFT
+        vDSP_fft_zrip(fftSetup!, &output, 1, log2n, FFTDirection(FFT_FORWARD))
+
+        // Parseval's theorem - Scale with respect to the number of bins
+        var scaledOutput = DSPSplitComplex(repeating: 0, count: inputCount)
+        var scaleMultiplier = DSPSplitComplex(repeating: 1.0 / Float(inputCount), count: 1)
+        vDSP_zvzsml(&output,
```

Comment:
Great question. @emurray2 do you have an opinion on this?

Reply:
Is it really not just:  c(a + bi) = ca + cbi

Reply:
Ahhh actually we are doing: (c + ci) * (a + bi) it seems...

Reply:
So we could create a complex representing (c + (0)i) and then perform the scaling. I actually had this by accident at first because I screwed up the DSPSplitComplex extension initializer and the scaling made a lot more sense with the reference value.

Reply:
https://github.com/AudioKit/AudioKit/pull/2582